src/topp/IDFilter.cpp

```diff
@@ -243,7 +245,38 @@ class TOPPIDFilter :
 
     vector<ProteinIdentification> proteins;
     vector<PeptideIdentification> peptides;
-    IdXMLFile().load(inputfile_name, proteins, peptides);
+
+    //only used for cxml
+    ConsensusMap cmap;
+    unordered_map<UInt64, ConsensusFeature*> id_to_featureref;
+
+    if (FileHandler::getTypeByFileName(inputfile_name) == FileTypes::IDXML)
+    {
+      IdXMLFile().load(inputfile_name, proteins, peptides);
+    }
+    else if (FileHandler::getTypeByFileName(inputfile_name) == FileTypes::CONSENSUSXML)
+    {
+      ConsensusXMLFile().load(inputfile_name, cmap);
+      for (auto& f : cmap)
+      {
+        UInt64 id = f.getUniqueId();
+        id_to_featureref[id] = &f;
+        for (auto&& p : f.getPeptideIdentifications())
```

Comment:
I find this confusing, since there is no case where we use reference proxies in our code (that I'm aware of).
Use `auto& p: ..`?